app/models/manageiq/providers/google/inventory/parser.rb

```diff
@@ -206,6 +206,23 @@ def instances
       instance_hardware(persister_vm, instance, flavor)
       instance_key_pairs(persister_vm, instance)
       instance_advanced_settings(persister_vm, instance)
+      vm_and_template_labels(persister_vm, instance.labels || [])
+    end
+  end
+
+  def vm_and_template_labels(resource, tags)
+    tags.each do |tag|
+      persister
+        .vm_and_template_labels
+        .find_or_build_by(
+          :resource => resource,
+          :name     => tag.first,
```

Comment:
Hm actually just noticed this .first and .second.

Is tags an array of arrays like `[[key1, value1], [key2, value2]]`?  Or is it a hash?

Reply:
(byebug) tag
[:"https-server", ""]
(byebug) tags
{:"http-server"=>"", :"https-server"=>""}

this is what I see in my breakpoint, so tags is a hash, and I guess when it iterates through the hash it converts it into an array?

Reply:
Yeah okay so it is a hash with `{key => value}`

Can you change the loop to `tags.each do |tag_name, tag_value|`

Reply:
Sure, I pushed it up. Is that what you're looking for?

Reply:
Yes looks good now thanks